plugins/inputs/http_listener_v2/http_listener_v2_test.go

```diff
@@ -0,0 +1,429 @@
+package http_listener_v2
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/influxdata/telegraf/testutil"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testMsg = "cpu_load_short,host=server01 value=12.0 1422568543702900257\n"
+
+	testMsgNoNewline = "cpu_load_short,host=server01 value=12.0 1422568543702900257"
+
+	testMsgs = `cpu_load_short,host=server02 value=12.0 1422568543702900257
+cpu_load_short,host=server03 value=12.0 1422568543702900257
+cpu_load_short,host=server04 value=12.0 1422568543702900257
+cpu_load_short,host=server05 value=12.0 1422568543702900257
+cpu_load_short,host=server06 value=12.0 1422568543702900257
+`
+	testJSONMsgs = `
```

Comment:
It's not often I get to say write less tests, but don't worry about testing different types of input data, these tests belong with the individual parsers.  The only cases to worry about are:
1. valid input
2. invalid (unparseable) input

Run all your tests against the influx parser.